### PR TITLE
NGG coding rework: Record shader stage handlers and cullers as members

### PR DESCRIPTION
**Pull request by amdrexu**
Don't use name finders to get shader stage handlers and cullers. Instead, record them as members.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 88ebe7d84c68b493629f029faafe49055fcf65d5
<details><summary>CTS tests (Failed: 0/211498)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39250/70506 (55.7%)</li>
<li>Failed: 0/70506 (0.0%)</li>
<li>Not Supported: 31256/70506 (44.3%)</li><li>Warnings: 0/70506 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39498/70495 (56.0%)</li>
<li>Failed: 0/70495 (0.0%)</li>
<li>Not Supported: 30997/70495 (44.0%)</li><li>Warnings: 0/70495 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36345/70497 (51.6%)</li>
<li>Failed: 0/70497 (0.0%)</li>
<li>Not Supported: 34152/70497 (48.4%)</li><li>Warnings: 0/70497 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/NggPrimShader.h [hunk 1]
```diff
@@ -355,9 +353,35 @@ class NggPrimShader {
...
     llvm::Value *vertexIndex2; // Relative index of vertex2 in NGG subgroup
-
   } m_nggInputs = {};
 
+  // ES handlers
+  struct {
+    llvm::Function *esMain; // ES main function
+    // A pair of part ES after ES splitting: the first one is to fetch cull data (position and cull distance) and the
+    // second one is to do deferred vertex exporting
+    std::pair<llvm::Function *, llvm::Function *> partEs;
```
Comment (nhaehnle) [inline lgc/patch/NggPrimShader.h:363]: Replace this with two members? `fetchCullData` and `deferredExports`, perhaps?

File: lgc/patch/NggPrimShader.h [hunk 2]
```diff
@@ -355,9 +353,35 @@ class NggPrimShader {
...
+  } m_esHandlers = {};
+
+  // GS handlers
+  struct {
+    llvm::Function *gsMain;     // GS main function
+    llvm::Function *copyShader; // Copy shader
+    llvm::Function *gsEmit;     // GS emit handler
+    llvm::Function *gsCut;      // GS cut handler
+  } m_gsHandlers = {};
```
Comment (nhaehnle) [inline lgc/patch/NggPrimShader.h:372]: Could remove the "gs" prefixes since it's already part of the containing struct name. Same for ES, actually.

**Review by nhaehnle — APPROVED** [review]
LGTM. Two nits inline.

File: lgc/patch/NggPrimShader.h (hunk 1 above)
Comment (amdrexu) [reply lgc/patch/NggPrimShader.h:363]: OK. I will make their names explicit.

File: lgc/patch/NggPrimShader.h (hunk 2 above)
Comment (amdrexu) [reply lgc/patch/NggPrimShader.h:372]: No problem.

**Comment by amdvlk-admin** [comment]
### Test summary for commit d24387c7d1ad6d72b06fb4add0d761c32b48f69d
<details><summary>CTS tests (Failed: 0/211487)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39248/70495 (55.7%)</li>
<li>Failed: 0/70495 (0.0%)</li>
<li>Not Supported: 31247/70495 (44.3%)</li><li>Warnings: 0/70495 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39498/70495 (56.0%)</li>
<li>Failed: 0/70495 (0.0%)</li>
<li>Not Supported: 30997/70495 (44.0%)</li><li>Warnings: 0/70495 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36345/70497 (51.6%)</li>
<li>Failed: 0/70497 (0.0%)</li>
<li>Not Supported: 34152/70497 (48.4%)</li><li>Warnings: 0/70497 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]
Thanks!

**Comment by amdvlk-admin** [comment]
### Test summary for commit d24387c7d1ad6d72b06fb4add0d761c32b48f69d
<details><summary>CTS tests (Failed: 0/211487)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39248/70495 (55.7%)</li>
<li>Failed: 0/70495 (0.0%)</li>
<li>Not Supported: 31247/70495 (44.3%)</li><li>Warnings: 0/70495 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39498/70495 (56.0%)</li>
<li>Failed: 0/70495 (0.0%)</li>
<li>Not Supported: 30997/70495 (44.0%)</li><li>Warnings: 0/70495 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36345/70497 (51.6%)</li>
<li>Failed: 0/70497 (0.0%)</li>
<li>Not Supported: 34152/70497 (48.4%)</li><li>Warnings: 0/70497 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**